### PR TITLE
Blocked Death & Instant Death Potion

### PR DESCRIPTION
**Pull request by ValenciaOrange**
I do not remember a issue number to go with this, But I'm sure that there is a lot of issues about the New Death Pots and Instant killer potion.
This should block/disable it.

### CODE REVIEW

**Comment by ValenciaOrange** [comment]
http://totalfreedom.boards.net/thread/26630/block-death-pots

**Comment by ValenciaOrange** [comment]
This Pull Request should solve #690 
and all of these http://i.imgur.com/TpIqxEG.png

File: src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java [hunk 1]
```diff
@@ -923,4 +930,45 @@ public void onPlayerLogin(PlayerLoginEvent event)
...
+            {
+                Entity e = event.getEntity();
+                if (e instanceof Player)
+                {
+                    Player player = (Player) e;
+                    playerMsg(player, "You are not permitted to use Death potions!", ChatColor.RED);
+                    event.setCancelled(true);
+                }
+            }
+
```
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java]: Blank like here, not sure why its there, but it shouldnt be.

File: src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java [hunk 2]
```diff
@@ -923,4 +930,45 @@ public void onPlayerLogin(PlayerLoginEvent event)
...
+        
+    @EventHandler    
+    public void onSplashPotion(PotionSplashEvent event)
+    {
+        for(PotionEffect effect : event.getPotion().getEffects())
+        {
+            if (effect.getType() == INVISIBILITY)
+            {
+                Entity e = event.getEntity();
+                if (e instanceof Player)
```
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java]: e is probably not the most descriptive way to word that...

**Comment by ValenciaOrange** [comment]
Removed the space and replaced:
Death Pots w/ entity d
Splash Invis w/ i

**Comment by PacksGamingHD** [comment]
This is from the old FOPM lol

**Comment by ValenciaOrange** [comment]
Bump

**Comment by tylerhyperHD** [comment]
I will say that it should not completely block the potion, but only block if it is at a certain potency/level.

File: src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java [hunk 3]
```diff
@@ -923,4 +930,44 @@ public void onPlayerLogin(PlayerLoginEvent event)
     {
         TFM_ServerInterface.handlePlayerLogin(event);
     }
+        
+    @EventHandler    
+    public void onSplashPotion(PotionSplashEvent event)
+    {
+        for(PotionEffect effect : event.getPotion().getEffects())
+        {
+            if (effect.getType() == INVISIBILITY)
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java]: These if statements are very repetitive. How about a fallthrough case statement instead?

File: src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java [hunk 4]
```diff
@@ -923,4 +930,44 @@ public void onPlayerLogin(PlayerLoginEvent event)
...
+                }
+            }
+            if (effect.getType() == HEAL)
+                //I beileve that Instant Health is called Heal in effect type??
+            {
+                Entity d = event.getEntity();
+                if (d instanceof Player)
+                {
+                    Player player = (Player) d;
+                    playerMsg(player, "You are not permitted to use Instant Death potions!", ChatColor.RED);
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java]: This isn't a death potion.

File: src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java [hunk 5]
```diff
@@ -923,4 +930,44 @@ public void onPlayerLogin(PlayerLoginEvent event)
...
+                }
+            }
+            if (effect.getType() == HEALTH_BOOST)
+                //Health Boost prevents player from respawning, Instant Health just instant kill
+            {
+                Entity d = event.getEntity();
+                if (d instanceof Player)
+                {
+                    Player player = (Player) d;
+                    playerMsg(player, "You are not permitted to use Death potions!", ChatColor.RED);
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java]: This isn't a death potion.

File: src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java [hunk 6]
```diff
@@ -923,4 +930,44 @@ public void onPlayerLogin(PlayerLoginEvent event)
...
+                Entity d = event.getEntity();
+                if (d instanceof Player)
+                {
+                    Player player = (Player) d;
+                    playerMsg(player, "You are not permitted to use Instant Death potions!", ChatColor.RED);
+                    event.setCancelled(true);
+                }
+            }
+            if (effect.getType() == HEALTH_BOOST)
+                //Health Boost prevents player from respawning, Instant Health just instant kill
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java]: Players can always use /suicide.

**Comment by AlexFreedomMod** [comment]
I think death potions are really high or low health potion

**Comment by Wild1145** [comment]
@JeromSar To re-open?

**Pull request closed without merging**